### PR TITLE
Expose comments through ParseUserDefinitionResult

### PR DESCRIPTION
**Pull request by VamsiModem**
Return comments in user definitions, so they can be colored properly

### CODE REVIEW

File: src/tests/Microsoft.PowerFx.Core.Tests/UserDefinedFunctionTests.cs [hunk 1]
```diff
@@ -153,5 +153,20 @@ public void TestCoercionWithUDFBody(string udfScript, string expectedIR)
...
+        public void TestCommentsFromUserDefinitionsScript(string script, int udfCount)
+        {
+            var parserOptions = new ParserOptions()
+            {
+                AllowsSideEffects = false
+            };
+
+            var parseResult = UserDefinitions.Parse(script, parserOptions);
+
+            Assert.Equal(udfCount, parseResult.Comments.Count());
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/UserDefinedFunctionTests.cs:169]: We should have at least one test where we validate the span of the comments to make sure that they are correct.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by CarlosFigueiraMSFT — APPROVED** [review]
🚢

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**